mapper/cumulocity/c8y_smartrest/src/smartrest_deserializer.rs

```diff
@@ -227,6 +228,27 @@ impl SmartRestLogRequest {
     }
 }
 
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+pub struct SmartRestRestartRequest {
+    pub message_id: String,
+    pub device: String,
+}
+
+impl SmartRestRestartRequest {
+    pub fn from_smartrest(smartrest: &str) -> Result<Self, SmartRestDeserializerError> {
+        let mut rdr = ReaderBuilder::new()
+            .has_headers(false)
+            .flexible(true)
+            .from_reader(smartrest.as_bytes());
+
+        match rdr.deserialize().next() {
+            Some(Ok(record)) => Ok(record),
+            Some(Err(err)) => Err(err)?,
+            None => panic!("empty request"),
```

Comment:
This code should not `panic`, maybe you need to use the same case as for `Some(Err...)`